src/sock_inet.c

```diff
@@ -491,6 +495,28 @@ static void sock_inet_prepare()
 		len = sizeof(val);
 		if (getsockopt(fd, IPPROTO_TCP, TCP_MAXSEG, &val, &len) == 0)
 			sock_inet6_tcp_maxseg_default = val;
+#endif
+		close(fd);
+	}
+
+	fd = socket(AF_INET, SOCK_STREAM, IPPROTO_MPTCP);
```

Comment:
you need to surround all this with:

```
#ifdef __linux__
(...)
#endif
```

Similar to what we are doing around proto_mptcpv[46], just to avoid doing a `socket(IPPROTO_MPTCP)` on Windows, where the behaviour is undefined I think (it should fail, but well, `IPPROTO_MPTCP` is not in POSIX standards)